File: snapchat/profile/oc/CodiaDemoOC/src/profileProfile/ProfileGroup7.m
//
//  ProfileGroup7.m
//
//  Created by codia-figma
//

#import <UIKit/UIKit.h>
#import <Masonry/Masonry.h>
#import "ProfileGroup7.h"
#import <UIKit/UIImageView.h>

@interface ProfileGroup7 ()

/// UIImageView: add
@property (nonatomic, strong) UIImageView *profileAdd;
@property (nonatomic, strong) UILabel *profileTextAddFriends;
/// UIImageView: back-R
@property (nonatomic, strong) UIImageView *profileImageBack;

@end

@implementation ProfileGroup7

- (instancetype)initWithFrame:(CGRect)frame {
    if (self = [super initWithFrame:frame]) {
        self.backgroundColor = [UIColor colorWithRed:1.00 green:1.00 blue:1.00 alpha:1.00];
        self.layer.cornerRadius = 10;
        self.layer.shadowOpacity = 1;
        self.layer.shadowRadius = 4;
        self.layer.shadowOffset = CGSizeMake(0, 2);
        self.layer.shadowColor = [UIColor colorWithRed:0.00 green:0.00 blue:0.00 alpha:0.25].CGColor;
        [self setupSubviews];
        [self setupConstraints];
    }
    return self;
}

- (void)layoutSubviews {
    [super layoutSubviews];
}


- (void)setupSubviews {
    [self addSubview:self.profileAdd];
    [self addSubview:self.profileTextAddFriends];
    [self addSubview:self.profileImageBack];
}

- (void)setupConstraints {
    [self.profileAdd mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.equalTo(self.mas_left).offset(14);
        make.width.mas_equalTo(20);
        make.top.equalTo(self.mas_top).offset(17);
        make.height.mas_equalTo(17.78);
    }];

    [self.profileTextAddFriends mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.equalTo(self.profileAdd.mas_right).offset(18);
        make.width.mas_equalTo(258);
        make.centerY.equalTo(self.profileAdd.mas_bottom).offset(-8.78);
    }];

    [self.profileImageBack mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.equalTo(self.profileTextAddFriends.mas_right).offset(9);
        make.width.mas_equalTo(15);
        make.centerY.equalTo(self.profileTextAddFriends.mas_centerY);
        make.height.mas_equalTo(15);
    }];
}




- (UIImageView *)profileAdd {
    if (!_profileAdd) {
        _profileAdd = [[UIImageView alloc] initWithFrame:CGRectZero];
        _profileAdd.image = [UIImage imageNamed:@"profileAdd_2441@2x.png"];
        _profileAdd.contentMode = UIViewContentModeScaleAspectFit;
    }
     return _profileAdd;
}

- (UILabel *)profileTextAddFriends {
    if (!_profileTextAddFriends) {
        _profileTextAddFriends = [[UILabel alloc] initWithFrame:CGRectZero];
        _profileTextAddFriends.text = @"Add Friends";
        _profileTextAddFriends.textAlignment = NSTextAlignmentLeft;
        _profileTextAddFriends.font = [UIFont fontWithName:@"Calibri" size:15] ?: [UIFont systemFontOfSize:15];
        _profileTextAddFriends.numberOfLines = 1;
        _profileTextAddFriends.textColor = [UIColor colorWithRed:0.00 green:0.00 blue:0.00 alpha:1.00];
    }
    return _profileTextAddFriends;
}

- (UIImageView *)profileImageBack {
    if (!_profileImageBack) {
        _profileImageBack = [[UIImageView alloc] initWithFrame:CGRectZero];
        _profileImageBack.image = [UIImage imageNamed:@"profileImageBack_502541@2x.png"];
        _profileImageBack.contentMode = UIViewContentModeScaleAspectFit;
    }
     return _profileImageBack;
}

@end

File: snapchat/profile/swift/CodiaDemoSwift/src/profileProfile/ProfileGroup7.swift
//
//  ProfileGroup7.swift
//
//  Created by codia-figma
//

import UIKit
import SnapKit


class ProfileGroup7: UIView {

    /// UIImageView: add
    lazy var profileAdd: UIImageView = {
        let view = UIImageView()
        view.image = UIImage(named: "profileAdd_2886@2x.png")
        view.contentMode = .scaleAspectFill

        return view
    }()

    lazy var profileTextAddFriends: UILabel = {
        let label = UILabel()
        label.text = "Add Friends"
        label.font = UIFont(name: "Calibri", size: 15) ?? UIFont.systemFont(ofSize: 15)
        label.textAlignment = .left
        label.textColor = UIColor(red: 0.00, green: 0.00, blue: 0.00, alpha: 1.00)
        label.numberOfLines = 1;
        return label
    }()

    /// UIImageView: back-R
    lazy var profileImageBack: UIImageView = {
        let view = UIImageView()
        view.image = UIImage(named: "profileImageBack_5032986@2x.png")
        view.contentMode = .scaleAspectFill

        return view
    }()

    override init(frame: CGRect) {
        super.init(frame: frame)
        self.backgroundColor = UIColor(red: 1.00, green: 1.00, blue: 1.00, alpha: 1.00)
        self.layer.cornerRadius = 10
        self.layer.shadowOpacity = 1
        self.layer.shadowRadius = 4
        self.layer.shadowOffset = CGSize(width: 0, height: 2)
        self.layer.shadowColor = UIColor(red: 0.00, green: 0.00, blue: 0.00, alpha: 0.25).cgColor
        setupSubviews()
        setupConstraints()
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    func layoutSubview() {
      super.layoutSubviews()
    }

    func setupSubviews() {
        self.addSubview(self.profileAdd)
        self.addSubview(self.profileTextAddFriends)
        self.addSubview(self.profileImageBack)
    }

    func setupConstraints() {
        profileAdd.snp.makeConstraints { make in
            make.left.equalTo(self.snp.left).offset(14)
            make.width.equalTo(20)
            make.top.equalTo(self.snp.top).offset(17)
            make.height.equalTo(17.78)
        }

        profileTextAddFriends.snp.makeConstraints { make in
            make.left.equalTo(profileAdd.snp.right).offset(18)
            make.width.equalTo(258)
            make.centerY.equalTo(profileAdd.snp.bottom).offset(-8.78)
        }

        profileImageBack.snp.makeConstraints { make in
            make.left.equalTo(profileTextAddFriends.snp.right).offset(9)
            make.width.equalTo(15)
            make.centerY.equalTo(profileTextAddFriends.snp.centerY)
            make.height.equalTo(15)
        }
    }
}